instrumentation/apolloconfig/apolloconfig-2.0.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apolloconfig/v2_0_0/ApolloRepositoryChangeInstrumentation.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apolloconfig.v2_0_0;
+
+import static io.opentelemetry.javaagent.bootstrap.Java8BytecodeBridge.currentContext;
+import static io.opentelemetry.javaagent.instrumentation.apolloconfig.v2_0_0.ApolloConfigSingletons.REPOSITORY_CHANGE_REPEAT_CONTEXT_KEY;
+import static io.opentelemetry.javaagent.instrumentation.apolloconfig.v2_0_0.ApolloConfigSingletons.instrumenter;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeTransformer;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+
+public class ApolloRepositoryChangeInstrumentation implements TypeInstrumentation {
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("com.ctrip.framework.apollo.internals.AbstractConfigRepository");
+  }
+
+  @Override
+  public void transform(TypeTransformer transformer) {
+    String adviceName = this.getClass().getName() + "$ApolloRepositoryChangeAdvice";
+    transformer.applyAdviceToMethod(named("fireRepositoryChange"), adviceName);
+  }
+
+  @SuppressWarnings("unused")
+  public static class ApolloRepositoryChangeAdvice {
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static void onEnter(
+        @Advice.Argument(value = 0) String namespace,
+        @Advice.Local("otelContext") Context context,
+        @Advice.Local("otelScope") Scope scope) {
+      Context parentContext = currentContext();
+      String repeat = parentContext.get(REPOSITORY_CHANGE_REPEAT_CONTEXT_KEY);
+      if (repeat != null) {
+        return;
+      }
+      if (!instrumenter().shouldStart(parentContext, namespace)) {
+        return;
+      }
+
+      context = instrumenter().start(parentContext, namespace);
```

Comment:
perhaps instead of spans this should be modeled as events?

Reply:
The logic of this library is roughly as follows

- watch ( http long pull. etc. )
- notify (Our current advise point)
   - listener1：spring env update
   - listener2：spring event
   - listener3：user custom，may have call 
   - ...

Reply:
Besides creating a span we could consider an event like in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/3fd27aa4d134d0f39bba5166eb94dc1b64bd9a1d/instrumentation/runtime-telemetry/runtime-telemetry-java8/javaagent/src/main/java/io/opentelemetry/instrumentation/javaagent/runtimemetrics/java8/JarAnalyzer.java#L242 @trask WDYT?

Reply:
an event sounds like a good option here

Reply:
I am not very familiar with event signals, which look like logs. I am not sure whether to remove span and replace it with event or add an event based on span.

If it is just one line of event log, I don't think it is helpful. The detailed background is as follows https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/12787#issuecomment-2530618737

Reply:
it looks like this has been added to today's Java APAC-friendly meeting, let's chat then